Introduce CMS Controller Manager (#3058)
kyma-project/kyma: components/cms-controller-manager/README.md

> # CMS Controller Manager

Reviewer: ```suggestion
# Content Management System (CMS) Controller Manager
```

## Overview

> CMS is a Kubernetes-native solution for storing managing content. It is build on top of Asset Store and it consists of the DocsTopic Controller and the ClusterDocsTopic Controller.

Reviewer: ```suggestion
CMS is a Kubernetes-native solution for managing content. It is built on top of the Asset Store and consists of the DocsTopic Controller and the ClusterDocsTopic Controller.
```

## Prerequisites

Use the following tools to set up the project:

* [Go distribution](https://golang.org)
* [Docker](https://www.docker.com/)
* [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)

## Usage

### Run a local version

To run the application outside the cluster, run this command:

```bash
make run
```

### Build a production version

To build the production Docker image, run this command:

```bash
IMG={image_name}:{image_tag} make docker-build
```

The variables are:

* `{image_name}` which is the name of the output image. Use `assetstore-controller-manager` for the image name.
* `{image_tag}` which is the tag of the output image. Use `latest` for the tag name.

> ### Environmental Variables

Reviewer: ```suggestion
### Environment Variables
```

Use the following environment variables to configure the application:

| Name | Required | Default | Description |
|------|----------|---------|-------------|

> | **APP_CLUSTER_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a ClusterDocsTopic CR |

Reviewer: ```suggestion
| **APP_CLUSTER_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a ClusterDocsTopic CR. |
```

> | **APP_CLUSTER_BUCKET_REGION** | No |  | Specifies the location of the region under which the controller creates the cluster bucket. If the field is empty, the controller creates the cluster bucket under the default location. |

Reviewer: ```suggestion
| **APP_CLUSTER_BUCKET_REGION** | No |  | Specifies the location of the region in which the controller creates a ClusterBucket CR. If the field is empty, the controller creates the bucket under the default location. |
```

| **APP_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a DocsTopic CR |

> | **APP_BUCKET_REGION** | No |  | Specifies the location of the region under which the controller creates the bucket. If the field is empty, the controller creates the bucket under the default location. |

Reviewer: ```suggestion
| **APP_BUCKET_REGION** | No |  | Specifies the location of the region in which the controller creates a Bucket CR. If the field is empty, the controller creates the bucket under the default location. |
```
